eventnotifier/slack/slack.go

```diff
@@ -96,105 +107,178 @@ func (n *Notifier) GetNotifierName() string {
-	headerText := utils.BuildHeaderMessageFromDisruptionEvent(dis, notifType)
-	headerBlock := slack.NewHeaderBlock(slack.NewTextBlockObject("plain_text", headerText, false, false))
-	bodyText := utils.BuildBodyMessageFromDisruptionEvent(dis, event, true)
-	bodyBlock := slack.NewSectionBlock(slack.NewTextBlockObject("mrkdwn", bodyText, false, false), nil, nil)
-	disruptionBlocks := n.buildSlackBlocks(dis, notifType)
+	return nil
+}
 
-	userInfo, err := dis.UserInfo()
-	if err != nil {
-		n.logger.Errorw("unable to retrieve disruption user info", "error", err, "disruption", dis.Name)
-	}
+func (n *Notifier) notifyForDisruption(dis *v1beta1.Disruption, event corev1.Event, notifType types.NotificationType) error {
+	logger := n.logger.With("disruptionName", dis.Name, "eventType", event.Type)
 
-	// Whenever a purpose is defined, we expect it to be available into all notifications sent messages
-	if nil != dis.Spec.Reporting && dis.Spec.Reporting.Purpose != "" {
-		disruptionBlocks = append(disruptionBlocks, slack.NewTextBlockObject("mrkdwn", "*Purpose:*\n"+dis.Spec.Reporting.Purpose, false, false))
-	}
+	slackMsg := n.buildSlackMessage(dis, event, notifType, dis.Spec.Reporting, logger)
 
 	if n.config.MirrorSlackChannelID != "" {
-		n.sendMessageToChannel(userInfo, n.config.MirrorSlackChannelID, headerText, headerBlock, disruptionBlocks, bodyBlock)
+		if err := n.sendMessageToChannel(n.config.MirrorSlackChannelID, slackMsg); err != nil {
+			logger.Warnw("slack notifier: couldn't send a message to the mirror slack channel", "slackChannel", n.config.MirrorSlackChannelID, "error", err)
+		}
 	}
 
 	if nil != dis.Spec.Reporting && dis.Spec.Reporting.SlackChannel != "" && dis.Spec.Reporting.MinNotificationType.Allows(notifType) {
-		n.sendMessageToChannel(userInfo, dis.Spec.Reporting.SlackChannel, headerText, headerBlock, disruptionBlocks, bodyBlock)
+		if err := n.sendMessageToChannel(dis.Spec.Reporting.SlackChannel, slackMsg); err != nil {
+			logger.Warnw("slack notifier: couldn't send a message to the channel from the reporting", "slackChannel", dis.Spec.Reporting.SlackChannel, "error", err)
+		}
 	}
 
 	// We expect notification equal to or above success to be sent to users
 	if !types.NotificationSuccess.Allows(notifType) {
-		n.logger.Debugw("slack notifier: not sending info notification type to not flood user", "disruptionName", dis.Name, "eventType", event.Type, "message", bodyText)
+		logger.Debugw("slack notifier: not sending info notification type to not flood user", "message", slackMsg.BodyText)
 
 		return nil
 	}
 
-	emailAddr, err := mail.ParseAddress(userInfo.Username)
-	if err != nil {
-		return fmt.Errorf("slack notifier: invalid user info email in disruption %s: %w", dis.Name, err)
+	if err := n.sendMessageToUserChannel(dis, slackMsg, logger); err != nil {
+		logger.Warnw("slack notifier: couldn't send a message to the user", "error", err)
```

Comment:
We don't need this logged warning. The error will eventually be returned all the way to client-go's `tools/record/event.go#recordEvent` and will be logged as an error